### ISSUE TITLE
avoid term 'lvalue' in error messages

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#5068]
### Summary of Problem

The compiler currently uses the term 'lvalue' in its error messages.  I think we should stop doing this.

### Steps to Reproduce

**Source Code:**

From test/variables/constants/assignConstError.chpl added in PR #5067 :

```chapel
const two = 2;
two = 3;
```

**Compile command:**
`chpl assignConstError.chpl`

**Execution command:**
N/A

### Configuration Information

- Output of `chpl --version`: 1.14.0.7d05050
- Output of `$CHPL_HOME/util/printchplenv --anonymize`: N/A
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`: N/A
- (For Cray systems only) Output of `module list`: N/A

**Comment by lydia-duncan** [comment]
I wholeheartedly support moving away from "lvalue" in error messages

**Comment by bradcray** [comment]
I threw the "easy" tag on here because I think it is (modulo agreeing on what the new error messages should say... but lots of things would be better than what we have today).

**Comment by dmk42** [comment]
I agree with the concept for general users, though seeing the term lvalue in an error message saved me some work the other day, since it pinpointed exactly what was wrong.  It will be difficult to come up with another term that works as well for that purpose.

**Comment by ben-albrecht** [comment]
I don't think this should have an `[easy]` label until we agree upon a replacement, which doesn't seem to be happening anytime soon.

**Comment by bradcray** [comment]
I think the person who owns this issue will need to sort through the error messages and propose alternatives based on the context (e.g., things like 'const', 'param', or 'read-only' may be appropriate depending on the context).  I still assert that this is an easy task (including the proposal) but I'm trying to hold restrain myself from getting caught up into a label wrestling war.
